data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -23,10 +23,16 @@ for name in ironic-api ironic-conductor ironic-inspector dnsmasq httpd mariadb i
 done
 
 # Start the provisioning nic if not already started
-PROVISIONING_NIC=ens4
-if ! nmcli -t device | grep "$PROVISIONING_NIC:ethernet:connected:provisioning"; then
+PROVISIONING_NIC={{.PlatformData.BareMetal.ProvisioningInterface}}
+
+if ! nmcli -t device | grep "$PROVISIONING_NIC:ethernet:connected"; then
     nmcli c add type ethernet ifname $PROVISIONING_NIC con-name provisioning {{ if .PlatformData.BareMetal.ProvisioningIPv6 }} ip6 {{ else }} ip4 {{ end }} {{.PlatformData.BareMetal.ProvisioningIP}}/{{.PlatformData.BareMetal.ProvisioningCIDR}}
     nmcli c up provisioning
+else
+  connection=$(nmcli -t device show $PROVISIONING_NIC | grep GENERAL.CONNECTION | cut -d: -f2)
+  nmcli con modify "$connection" ifname $PROVISIONING_NIC {{ if .PlatformData.BareMetal.ProvisioningIPv6 }} ip6 {{ else }} ip4 {{ end }} {{.PlatformData.BareMetal.ProvisioningIP}}/{{.PlatformData.BareMetal.ProvisioningCIDR}}
```

Comment:
This is used to add the provisioning IP to an existing interface, when it's on the baremetal network.

Reply:
I was thinking we'd not have any provisioning IP at all in this case, and use the IP on the baremetal network - do we need this so the IP is more easily predictable prior to the VM booting?

Reply:
Added some ideas around this to https://github.com/openshift-metal3/dev-scripts/pull/1069 for discussion